silkworm/rpc/core/estimate_gas_oracle.cpp

```diff
@@ -89,13 +89,12 @@ Task<intx::uint256> EstimateGasOracle::estimate_gas(const Call& call, const silk
             EVMExecutor executor{config_, workers_, state};
             auto mid = (hi + lo) / 2;
             transaction.gas_limit = mid;
-
             result = try_execution(executor, block, transaction);
             if (result.success()) {
                 hi = mid;
             } else {
                 lo = mid;
-                if (result.pre_check_error == std::nullopt) {
+                if (result.pre_check_error_code != std::nullopt && *(result.pre_check_error_code) != PreCheckErrorCode::EC_INTRINSIC_GAS_TOO_LOW) {
```

Comment:
Probably this could be simplified to: `result.pre_check_error_code && result.pre_check_error_code != ...`

Reply:
ok